### ISSUE TITLE
[Bug] DeepOmit not working as expected

### ISSUE THREAD

**Issue by FlynnHillier** [tobloef/deep-utility-types#3]
using this type
```typescript
export type GAMECONTEXT = {
  game?: {
    id: UUID;
    players: Partial<BW<Player>>;
    moves: VerboseMovement[];
    viewing?: {
      move: VerboseMovement;
      index: number;
      isLatest: boolean;
    };
    time: {
      start: number;
    };
    live?: {
      time: {
        lastUpdated: number;
        remaining?: BW<number>;
      };
      current: {
        fen: string;
        turn: Color;
      };
      engine: {
        getValidMoves: Chess["moves"];
      };
    };
  };
  conclusion?: {
    victor?: Color;
    reason: GameTermination;
  };
};
```

I am trying to omit the type `"game.live.engine.getValidMoves"` however - using `DeepOmit<GAMECONTEXT, "game.live.engine.getValidMoves">` does not omit the value specified, it is still required on the specified type.

Is this potentially because some of my nested objects are marked as optional?

![image](https://github.com/tobloef/deep-utility-types/assets/48843724/96e92d57-25fe-4346-b45a-c130a87a9020)
![image](https://github.com/tobloef/deep-utility-types/assets/48843724/c8156a02-85ef-42aa-a892-d8d37b9baf36)

**Comment by tobloef** [comment]
I'll take a look when I have a moment, thanks!

**Comment by FlynnHillier** [comment]
> I'll take a look when I have a moment, thanks!

Thanks for the quick reply - let me know if you have any trouble recreating the issue :)

**Comment by tobloef** [comment]
I believe I have fixed this in ab7696b. Could you try updating to the new `v1.3.1` and see if it works correctly now?

**Comment by FlynnHillier** [comment]
> I believe I have fixed this in [ab7696b](https://github.com/tobloef/deep-utility-types/commit/ab7696b52f02e9f51a3e7562d0fc1032ad90210f). Could you try updating to the new `v1.3.1` and see if it works correctly now?

Just upgraded and the `DeepOmit` type seems to be working as expected now 👍 Thanks!
